analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/IOperationExtensionsTest.cs

```diff
@@ -152,32 +151,58 @@ public void TrackedSymbol_DeclarationExpression(string assignment)
     {
         var code = $"public class C {{ void Method() {{ {assignment}; }} }}";
         var graph = TestHelper.CompileCfgCS(code);
-        var allDeclarations = graph.Blocks[1].Operations.SelectMany(x => x.DescendantsAndSelf()).Where(x => x.Kind == OperationKindEx.DeclarationExpression).Select(IDeclarationExpressionOperationWrapper.FromOperation).ToArray();
+        var allDeclarations = graph.Blocks[1].Operations
+            .SelectMany(x => x.DescendantsAndSelf()).Where(x => x.Kind == OperationKindEx.DeclarationExpression).Select(IDeclarationExpressionOperationWrapper.FromOperation).ToArray();
         allDeclarations.Should().NotBeEmpty();
         allDeclarations.Should().AllSatisfy(x =>
-            x.WrappedOperation.TrackedSymbol().Should().NotBeNull().And.BeAssignableTo<ISymbol>()
+            x.WrappedOperation.TrackedSymbol(ProgramState.Empty).Should().NotBeNull().And.BeAssignableTo<ISymbol>()
             .Which.GetSymbolType().Should().NotBeNull().And.BeAssignableTo<ITypeSymbol>()
             .Which.SpecialType.Should().Be(SpecialType.System_Int32));
     }
 
+    [TestMethod]
+    public void TrackedSymbol_CaptureReference()
+    {
+        var cfg = TestHelper.CompileCfgBodyCS("a ??= b;", "object a, object b");
+        var capture = IFlowCaptureOperationWrapper.FromOperation(cfg.Blocks[1].Operations[0]);
+        var captureReference = cfg.Blocks[3].Operations[0].ChildOperations.First();
+        var state = ProgramState.Empty.SetCapture(capture.Id, capture.Value);
+        captureReference.TrackedSymbol(state).Name.Should().Be("a");
```

Comment:
This would throw NRE if it would return `null`. There should be something like
```
captureReference.TrackedSymbol(state).Should().NotBeNull().And.BeAssignableTo<ISymbol>().Which.Name.Should().Be("a");
```
I don't like the `BeAssignmableTo` part, but for some weird reason, I can't get it work without it

Reply:
Similar below

Reply:
I think this is nice:
`(conversion.TrackedSymbol(state)?.Name).Should().Be("a");`

Reply:
Isn't the `?` optimize out the `Should()` call?

Reply:
That's what the `(...)` are for!

Reply:
I would never guessed that. Let's make the UT explicit to be clear about what it does.